api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto

```diff
@@ -382,7 +382,16 @@ message CheckSettings {
   //   :ref:`grpc_service<envoy_v3_api_field_extensions.filters.http.ext_authz.v3.ExtAuthz.grpc_service>`.
   map<string, string> context_extensions = 1 [(udpa.annotations.sensitive) = true];
 
-  // When set to true, disable the configured :ref:`with_request_body
-  // <envoy_v3_api_field_extensions.filters.http.ext_authz.v3.ExtAuthz.with_request_body>` for a route.
-  bool disable_request_body_buffering = 2;
+  oneof override {
+    option (validate.required) = true;
+
+    // When set to true, disable the configured :ref:`with_request_body
+    // <envoy_v3_api_field_extensions.filters.http.ext_authz.v3.ExtAuthz.with_request_body>` for a specific route.
+    bool disable_request_body_buffering = 2;
+
+    // Enable or override request body buffering, which is configured with
+    // :ref:`with_request_body <envoy_v3_api_field_extensions.filters.http.ext_authz.v3.ExtAuthz.with_request_body>`
+    // for a specific route.
+    BufferSettings with_request_body = 3;
+  }
```

Comment:
Please don't move the `disable_request_body_buffering` to a oneof directly. You can add a new `with_request_body`  field only and document it clearly only one field could be set or valid clearly.

Reply:
Removed the `oneof` and added a validation check instead.